res/controllers/MVave-SMK-25-II-scripts.js

```diff
@@ -0,0 +1,277 @@
+            });
+            this.forwardButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0x90, 0x5C],
+                key: "beatjump_forward",
+            });
+            // Faders
+            this.pitchStrip = new components.Pot({
+                group: "[Channel1]",
+                midi: [0xE0, 0xE0],
+                key: "rate",
+            });
+            this.modStrip = new components.Pot({
+                group: "[Channel1]",
+                midi: [0xB0, 0x01],
+                key: "volume",
+            });
+        }
+    }
+
+    class Controller extends components.ComponentContainer {
+        constructor() {
+            super({});
+            this.activeDeck = new Deck();
+            this.gainKnob = new components.Encoder({
+                group: "[Master]",
+                midi: [0xB0, 0x25],
+                key: "gain",
+            });
+            this.headGainKnob = new components.Encoder({
+                group: "[Master]",
+                midi: [0xB0, 0x24],
+                key: "headGain",
+            });
+            this.xfadeKnob = new components.Encoder({
+                group: "[Master]",
+                midi: [0xB0, 0x23],
+                key: "crossfader",
+            });
+            this.recordButton = new components.Button({
+                group: "[Recording]",
+                midi: [0x90, 0x5F],
+                inKey: "toggle_recording",
+                outKey: "status",
+            });
+            // Deck Select
+            this.deckLeftButton = new components.Button({
+                group: "[Channel1]",
+                midi: [0x90, 0x2E], // << Channel Left Button
+                inToggle: function() {
+                    SMK25II.controller.activeDeck.setCurrentDeck("[Channel1]");
+                },
+            });
+            this.deckRightButton = new components.Button({
+                group: "[Channel2]",
+                midi: [0x90, 0x2F], // >> Channel Right button
+                inToggle: function() {
+                    SMK25II.controller.activeDeck.setCurrentDeck("[Channel2]");
+                },
+            });
+            // Drum Pads
+            this.samplers = [];
+            for (let i = 0; i < 16; i++) {
+                this.samplers[i] = new components.SamplerButton({
+                    number: i + 1,
+                    volumeByVelocity: true,
+                });
+            }
+        }
+    }
+
+    SMK25II.init = function() {
+        SMK25II.controller = new Controller();
+    };
+    SMK25II.shutdown = function() {
+        SMK25II.controller.shutdown();
+    };
+
+    const MMCHeader = [0xF0, 0x35, 0x59];
+
+    /**
+     *
+     * @param data the full sysex message data.
+     */
+    const decodeButton = function(data) {
+        // Hotcues
+        if (data[4] !== undefined && data[4] >= 0x00 && data[4] <= 0x07) {
+            SMK25II.controller.activeDeck.hotcues[data[4]].inToggle();
```

Comment:
the standard way would be to map `input`, not `inToggle` directly.

Reply:
Also, does it really not send a release message?

Reply:
It does send a release message, this is probably toggling on both now that I look at it again, I'll swap it to "input" and check the value.

Reply:
Actually, that's not true, they appear to send the release message immediately even if you're still pressing it. Oh well, a quirk of the firmware I suppose. I'll leave a comment.

Reply:
If I change this to "input" it will break the deck left and deck right buttons because I overwrote inToggle for those to handle swapping the active deck. I thought this is what I did in the other mapping too, but they stop working if I call input. Should I be overriding and calling something else?

Reply:
Actually, that makes perfect sense, inToggle only worked in the other plugin because it was a power-window button. Fixed!